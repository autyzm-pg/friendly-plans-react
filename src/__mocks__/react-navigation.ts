const createAppContainer = (component: any) => component;
const createStackNavigator = jest.fn();
const createDrawerNavigator = jest.fn();
const createBottomTabNavigator = jest.fn();
const createSwitchNavigator = jest.fn();
const withNavigation = jest.fn();

const NavigationActions = {
  navigate: jest.fn(),
};

const SafeAreaView = {
  setStatusBarHeight: jest.fn(),
};

export {
  createAppContainer,
  createStackNavigator,
  createBottomTabNavigator,
  createSwitchNavigator,
  createDrawerNavigator,
  NavigationActions,
  SafeAreaView,
  withNavigation,
};
