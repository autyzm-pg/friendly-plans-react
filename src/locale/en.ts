const translations = {
  planActivity: {
    planNamePlaceholder: 'Type plan name',
    shuffleTasks: 'Shuffle tasks',
  },
  common: {
    cancel: 'Cancel',
    ok: 'OK',
    yes: 'Yes',
    email: 'Email address',
    name: 'Your Name',
    password: 'Password',
    error: 'Error',
    success: 'Success',
    unknownError: 'Unknown error occured',
    addImage: 'Add Image',
  },
  validation: {
    email: 'Must be a valid email',
    required: 'This field is required',
    passwordLength: 'Password is too short',
    acceptRequired: 'You need to accept our Terms of Use',
    duplicatedPlan: 'Plan name already exists',
    planNameRequired: 'Plan name is required',
  },
  notifications: {
    channelName: 'Notifications',
    channelDescription: 'Get urgent notifications about the service',
  },
  signIn: {
    signIn: 'Sign In',
    signInButton: 'Sign In',
    signInAsAnonymous: 'Use the App as Guest',
    signUpTip: "don't have an account?",
    anonymousTip: 'you can also',
    forgotPassword: 'forgot password?',
  },
  signUp: {
    signUp: 'Sign Up',
    signUpButton: 'Sign up',
    accept: 'I accept ',
    termsOfUse: 'Terms of Use',
  },
  resetPassword: {
    resetPassword: 'Reset password',
    guide1: 'Provide you email in the field below. We will send you link to reset your password.',
    guide2: 'After setting up new password, sign in with newly set password at the login screen.',
    resetPasswordButtonLabel: 'Reset Password',
    userNotFound: 'No user found for provided email. Make sure to provide email you registered.',
    resetPasswordSuccess:
      'Email with password reset link has been sent. After setting up new password, sign in with newly set password at the login screen.',
  },
  content: {
    content: 'Content',
    contentList: 'Content List',
  },
  settings: {
    settings: 'Settings',
    signOutTitle: 'Sign out',
    signOutSubtitle: 'After you sign out, you will need to sign in again.',
    signOutAction: 'Sign out',
    signOutDialogTitle: 'Sign out',
    signOutDialogDescription: 'Are you sure you want to sign out?',
  },
  sidebar: {
    addTask: 'Add SimpleTask',
    addStudent: 'Add Student',
    takeAPicture: 'Take a Picture',
    recordSound: 'Record Sound',
  },
  studentList: {
    createStudent: 'Add a Student',
    screenTitle: 'Select student',
    search: 'Search',
    dashboard: 'Student dashboard',
    studentNamePlaceholder: 'Enter Student name...',
    removeStudentTitle: 'Confirm',
    removeStudentDescription:
      'Are you sure you want to remove {{name}} from your student list?\n This action cannot be undone.',
  },
  planList: {
    viewTitle: 'All Plans',
    createPlan: 'Add new plan',
    copyPlan: 'Copy existing plan',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Task',
    viewTitleInteraction: 'Interaction',
    viewTitleBreak: 'Break',
    taskNamePlaceholder: 'Type task name',
    timerButton: 'Set up timer',
    taskNameForChild: 'Type task name for child',
    addImage: 'Add image',
    imageActionTakePhoto: 'Take a photo',
    imageActionLibrary: 'Browse from library',
    imageActionBrowse: 'Browse from device',
    imageLibraryTitle: 'Image Library',
  },
  updatePlan: {
    screenTitle: '{{studentName}} - Update a plan',
    removePlanTitle: 'Confirm',
    removePlanDescription:
      'Are you sure you want to plan "{{name}}" from student plan list?\n This action cannot be undone.',
    addBreak: 'Add break',
    addInteraction: 'Add interaction',
    addTask: 'Add task',
    planItemNamePlaceholder: 'Enter Plan Item name...',
  },
  studentSettings: {
    studentName: "Student's name",
    taskView: "Task's view",
    soundSettings: 'Sound settings',
    settingsTitle: "Student's settings",
    createStudentTitle: 'Create a Student',
    textSettingsSizeS: 'Font size: S',
    textSettingsSizeM: 'Font size: M',
    textSettingsSizeL: 'Font size: L',
    textSettingsSizeXL: 'Font size: XL',
    largeImageSlide: 'Large image as a slide',
    imageWithTextSlide: 'Image with text label as a slide',
    textSlide: 'Just text as a slide',
    imageWithTextList: 'image with text label as a list',
    textList: 'Just text label as a list',
    uppercase: 'Uppercase letters',
    blockSwipe: 'Block swipe',
    alarmSound: 'Alarm sound',
    planCardPlacehorder: "Let's play paper, rock, scissors",
    studentNamePlaceholder: 'Enter student name',
    removeStudent: 'Remove student',
    createStudent: 'Create student',
  },
  runPlan: {
    next: 'Next',
    wait: 'Wait...',
  },
  taskTable: {
    number: '#',
    name: 'Name',
    type: 'Type',
    section: 'Parts',
    time: 'Time',
    delete: 'Delete',
    edit: 'Edit',
  },
};

export default translations;
