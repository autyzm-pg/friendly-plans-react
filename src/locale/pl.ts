const translations = {
  planActivity: {
    planNamePlaceholder: 'Wpisz nazwę planu',
    shuffleTasks: 'Tasuj zadania',
  },
  common: {
    cancel: 'Anuluj',
  },
  signIn: {
    signIn: 'Zaloguj się',
    signInAsAnonymous: 'Zaloguj się anonimowo',
  },
  content: {
    content: 'Treść',
    contentList: 'Treści',
  },
  settings: {
    settings: 'Ustawienia',
    signOutTitle: 'Wyloguj się',
    signOutSubtitle: 'Po wylogowaniu się, musisz zalogować się ponownie',
    signOutAction: 'Wyloguj',
    signOutDialogTitle: 'Wyloguj się',
    signOutDialogDescription: 'Czy na pewno chcesz się wylogować?',
  },
  planList: {
    viewTitle: 'Wszystkie plany',
    createPlan: 'DODAJ NOWY PLAN',
    copyPlan: 'SKOPIUJ ISTNIEJĄCY PLAN',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Zadanie',
    viewTitleInteraction: 'Interakcja',
    viewTitleBreak: 'Przerwa',
    taskNamePlaceholder: 'Wpisz nazwę zadania',
    timerButton: 'Ustaw timer',
    taskNameForChild: 'Nazwij dla dziecka',
    addImage: 'Dodaj obrazek',
    imageActionTakePhoto: 'Zrób zdjęcie',
    imageActionLibrary: 'Wybierz z biblioteki',
    imageActionBrowse: 'Dodaj z urządzenia',
    imageLibraryTitle: 'Biblioteka zdjęć',
  },
  updatePlan: {
    addBreak: 'Dodaj przerwę',
    addInteraction: 'Dodaj interakcję',
    addTask: 'Dodaj zadanie',
  },
  studentSettings: {
    studentName: 'Imię podopiecznego',
    taskView: 'Widok zadań',
    soundSettings: 'Ustawienia dźwięku',
    settingsTitle: 'Ustawienia podopiecznego',
    createStudentTitle: 'Stwórz Studenta',
    textSettingsSizeS: 'Rozmiar liter: S',
    textSettingsSizeM: 'Rozmiar liter: M',
    textSettingsSizeL: 'Rozmiar liter: L',
    textSettingsSizeXL: 'Rozmiar liter: XL',
    largeImageSlide: 'Duży obrazek jako slajd',
    imageWithTextSlide: 'Obrazek z tekstem jako slajd',
    textSlide: 'Tylko tekst jako slajd',
    imageWithTextList: 'Obrazek z tekstem jako lista',
    textList: 'Tylko tekst jako lista',
    uppercase: 'Wielkie litery',
    blockSwipe: "Blokada swipe'u",
    alarmSound: 'Dźwięk alarmu',
    planCardPlacehorder: 'Zagraj w papier, kamień, nożyce',
    studentNamePlaceholder: 'Wpisz imię podopiecznego',
    removeStudent: 'Usuń studenta',
    createStudent: 'Dodaj Studenta',
  },
  studentsList: {
    screenTitle: 'Wybór podopiecznego',
    search: 'Wyszukaj',
    dashboard: 'Studenci',
    createStudent: 'Stwórz studenta',
  },
  runPlan: {
    next: 'Dalej',
    wait: 'Czekaj...',
  },
  taskTable: {
    number: '#',
    name: 'Nazwa',
    type: 'Typ',
    section: 'Części',
    time: 'Czas',
    delete: 'Usuń',
    edit: 'Edycja',
  },
  validation: {
    duplicatedPlan: 'Plan o podanej nazwie już istnieje',
    planNameRequired: 'Nazwa planu jest wymagana',
  },
};

export default translations;
