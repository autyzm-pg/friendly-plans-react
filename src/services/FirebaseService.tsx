import firebase from 'react-native-firebase';
import { Image } from 'react-native-image-crop-picker';

export class FirebaseService {
  /* Upload image to Firebase Storage and return public URL to download it. */
  static uploadUserImage = async (image: string): Promise<string> => {
    const user = firebase.auth().currentUser!;
    const imageRef = firebase
      .storage()
      .ref('users')
      .child(user.uid);
    const snapshot = await imageRef.putFile(image);
    return snapshot.downloadURL as string;
  };

  static updateUserImage = async (image: string): Promise<void> => {
    const photoURL = await FirebaseService.uploadUserImage(image);
    return firebase.auth().currentUser!.updateProfile({ photoURL });
  };
}
