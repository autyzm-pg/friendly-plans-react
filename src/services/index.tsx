import _AnalyticsService from './AnalyticsService';
import _CrashlyticsService from './CrashlyticsService';
import _NavigationService from './NavigationService';

export const NavigationService = new _NavigationService();
export const AnalyticsService = new _AnalyticsService(NavigationService);
export const CrashlyticsService = new _CrashlyticsService();
