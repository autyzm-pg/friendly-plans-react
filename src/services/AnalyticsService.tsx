import firebase from 'react-native-firebase';
import { NavigationState } from 'react-navigation';

import { User } from 'models';
import NavigationService from './NavigationService';

export default class AnalyticsService {
  navigationService: NavigationService;
  user: User | null;
  screenName?: string;

  constructor(navigationService: NavigationService) {
    firebase.analytics().setAnalyticsCollectionEnabled(!__DEV__);
    this.navigationService = navigationService;
    this.user = null;
  }

  setCurrentScreen = (prevState: NavigationState, currentState: NavigationState) => {
    const currentScreen = this.navigationService.getActiveRouteName(currentState);
    const prevScreen = this.navigationService.getActiveRouteName(prevState);

    if (prevScreen !== currentScreen) {
      firebase.analytics().setCurrentScreen(currentScreen, currentScreen);
      this.screenName = currentScreen;
    }
  };

  setCurrentUser = (user: User | null) => {
    this.user = user;
    firebase.analytics().setUserId(this.user ? this.user!.id.toString() : null);
  };
}
