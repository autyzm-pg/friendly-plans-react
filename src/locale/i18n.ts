import i18n, { LanguageDetectorModule } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { getLocales } from 'react-native-localize';

import en from './en';
import pl from './pl';

const languageDetector: LanguageDetectorModule = {
  type: 'languageDetector',
  init(services, detectorOptions, i18nextOptions) {
    /* use services and options */
  },
  detect() {
    return getLocales()[0].languageCode;
  },
  cacheUserLanguage(lng) {
    /* cache language */
  },
};

i18n
  .use(languageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    resources: {
      en,
      pl,
    },
    ns: ['common', 'signIn', 'signUp'],
    defaultNS: 'common',
    debug: __DEV__,
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
