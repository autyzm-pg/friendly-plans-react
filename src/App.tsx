import React, { PureComponent } from 'react';
import { I18nextProvider } from 'react-i18next';
import { Platform, StatusBar } from 'react-native';
import { createAppContainer, NavigationContainerComponent, SafeAreaView } from 'react-navigation';

import { i18n } from 'locale';
import { AnalyticsService, NavigationService } from 'services';
import { statusBarHeight } from 'styles';
import { RootStackNavigator } from './navigators/RootStackNavigator';

// Set status bar height on Android to support windowTranslucentStatus style
/* istanbul ignore next */
if (Platform.OS === 'android') {
  // @ts-ignore
  SafeAreaView.setStatusBarHeight(statusBarHeight);
}

const AppContainer = createAppContainer(RootStackNavigator);

export default class App extends PureComponent {
  setNavigationRef = (ref: NavigationContainerComponent) => {
    NavigationService.setTopLevelNavigator(ref);
  };

  render() {
    return (
      <I18nextProvider i18n={i18n}>
        <StatusBar hidden />
        <AppContainer ref={this.setNavigationRef} onNavigationStateChange={AnalyticsService.setCurrentScreen} />
      </I18nextProvider>
    );
  }
}
