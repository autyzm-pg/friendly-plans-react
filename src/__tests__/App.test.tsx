import { shallow } from 'enzyme';
import React from 'react';
import { NavigationContainerComponent } from 'react-navigation';

import * as locale from 'locale';
import { NavigationService } from 'services';
import App from '../App';

// @ts-ignore
locale.i18n = jest.fn();

describe('App', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set top level navigator ref', () => {
    const wrapper = shallow(<App />);
    const instance = wrapper.instance() as App;
    const navigator = {};
    const spy = jest
      .spyOn(NavigationService, 'setTopLevelNavigator')
      .mockImplementationOnce(() => null);
    instance.setNavigationRef(navigator as NavigationContainerComponent);
    expect(spy).toHaveBeenCalledWith(navigator);
  });
});
