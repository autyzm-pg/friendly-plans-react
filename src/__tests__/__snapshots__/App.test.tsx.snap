// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<I18nextProvider
  i18n={[MockFunction]}
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Component
    onNavigationStateChange={[Function]}
  />
</I18nextProvider>
`;
